descqa/readiness_test.py

```diff
@@ -123,30 +124,104 @@ def __init__(self, **kwargs):
         if not all(d.get('quantity') for d in self.uniqueness_to_check):
             raise ValueError('yaml file error: `quantity` must exist for each item in `uniqueness_to_check`')
 
+        self.enable_individual_summary = bool(kwargs.get('enable_individual_summary', True))
+        self.enable_aggregated_summary = bool(kwargs.get('enable_aggregated_summary', False))
+        self.always_show_plot = bool(kwargs.get('always_show_plot', True))
+
         self.nbins = int(kwargs.get('nbins', 50))
-        self.prop_cycle = cycle(iter(plt.rcParams['axes.prop_cycle']))
+        self.prop_cycle = None
+
+        self.current_catalog_name = None
+        self.current_failed_count = None
+        self._aggregated_header = list()
+        self._aggregated_table = list()
+        self._individual_header = list()
+        self._individual_table = list()
+
         super(CheckQuantities, self).__init__(**kwargs)
 
+    def record_result(self, results, quantity_name=None, more_info=None, failed=None, individual_only=False):
+        if isinstance(results, dict):
+            self.current_failed_count += sum(1 for v in results.values() if v[1] == 'fail')
+        elif failed:
+            self.current_failed_count += 1
+
+        if self.enable_individual_summary:
+            if quantity_name is None:
+                self._individual_header.append(self.format_result_header(results, failed))
+            else:
+                self._individual_table.append(self.format_result_row(results, quantity_name, more_info))
+
+        if self.enable_aggregated_summary and not individual_only:
+            if quantity_name is None:
+                results = '{} {}'.format(self.current_catalog_name, results) if self.current_catalog_name else results
+                self._aggregated_header.append(self.format_result_header(results, failed))
+            else:
+                quantity_name = '{} {}'.format(self.current_catalog_name, quantity_name) if self.current_catalog_name else quantity_name
+                self._aggregated_table.append(self.format_result_row(results, quantity_name, more_info))
 
-    def _format_row(self, quantity, plot_filename, results):
-        output = ['<tr>', '<td title="{1}">{0}</td>'.format(quantity, plot_filename)]
+    def format_result_row(self, results, quantity_name, more_info):
+        more_info = 'title="{}"'.format(more_info) if more_info else ''
+        output = ['<tr>', '<td {1}>{0}</td>'.format(quantity_name, more_info)]
         for s in self.stats:
             output.append('<td class="{1}" title="{2}">{0:.4g}</td>'.format(*results[s]))
         output.append('</tr>')
         return ''.join(output)
 
+    @staticmethod
+    def format_result_header(results, failed=False):
+        return '<span {1}>{0}</span>'.format(results, 'class="fail"' if failed else '')
+
+    def generate_summery(self, output_dir, aggregated=False):
```

Comment:
Should this method be named `generate_summary` (note the spelling error in "summary" in the code)?